packages/quantic/force-app/main/default/lwc/quanticResultTemplate/__tests__/quanticResultTemplate.test.js

```diff
@@ -0,0 +1,54 @@
+// @ts-ignore
+import QuanticResultTemplate from '../quanticResultTemplate';
+import {buildCreateTestComponent, cleanup, flushPromises} from 'c/testUtils';
+
+const createTestComponent = buildCreateTestComponent(
+  QuanticResultTemplate,
+  'c-quantic-result-template'
+);
+
+const selectors = {
+  slotByName: (name) => `slot[name="${name}"]`,
+};
+
+const slots = {
+  LABEL: 'label',
+  BADGES: 'badges',
+  ACTIONS: 'actions',
+  DATE: 'date',
+  VISUAL: 'visual',
+  TITLE: 'title',
+  METADATA: 'metadata',
+  EMPHASIZED: 'emphasized',
+  EXCERPT: 'excerpt',
+  BOTTOM_METADATA: 'bottom-metadata',
+  CHILDREN: 'children',
+};
+
+describe('c-quantic-result-template', () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  it('should display all slots by default', async () => {
+    const element = createTestComponent();
+    await flushPromises();
+
+    for (const [key, name] of Object.entries(slots)) {
+      let slot = element.shadowRoot.querySelectorAll(
+        selectors.slotByName(name)
+      );
+      expect(slot).not.toBeNull();
+      expect(slot.length).toBe(1);
+
+      const newElem = document.createElement('div');
+      newElem.setAttribute('slot', name);
+      newElem.innerText = key;
+      element.shadowRoot.appendChild(newElem);
+
+      slot = element.shadowRoot.querySelectorAll(selectors.slotByName(name));
+      expect(slot).not.toBeNull();
+      expect(slot.length).toBe(1);
```

Comment:
Not sure I get what this tests?

Is it really that you are appending a `<div slot={name}>{key}</div>` and then the selector `slot[name={name}]` actually works and will give you that `<div>` ?

If that's the case, I would also get the innerHTML of that and test it against the div you've made?

```
expect(slot[0].innerHTML).toEqual(`<div>${key}</div>`);
```

Would that work?

Reply:
Yea I agree but what does the `element.shadowRoot.appendChild` do?